pkg/runtime/runtime.go

```diff
@@ -2986,42 +2981,61 @@ func (a *DaprRuntime) createAppChannel() (err error) {
 
 // Returns the HTTP endpoint for the app.
 func (a *DaprRuntime) getAppHTTPEndpoint() string {
+	// Application protocol is "http" or "https"
 	port := strconv.Itoa(a.runtimeConfig.ApplicationPort)
-	if a.runtimeConfig.AppSSL {
-		return "https://" + channel.DefaultChannelAddress + ":" + port
-	} else {
+	switch a.runtimeConfig.ApplicationProtocol {
+	case HTTPProtocol, H2CProtocol:
 		return "http://" + channel.DefaultChannelAddress + ":" + port
+	case HTTPSProtocol:
+		return "https://" + channel.DefaultChannelAddress + ":" + port
+	default:
+		return ""
 	}
 }
 
 // Initializes the appHTTPClient property.
 func (a *DaprRuntime) initAppHTTPClient() {
-	var tlsConfig *tls.Config
-	if a.runtimeConfig.AppSSL {
-		tlsConfig = &tls.Config{
-			//nolint:gosec
-			InsecureSkipVerify: true,
-			// For 1.11
-			MinVersion: channel.AppChannelMinTLSVersion,
+	var transport nethttp.RoundTripper
+	if a.runtimeConfig.ApplicationProtocol == H2CProtocol {
+		// Enable HTTP/2 Cleartext transport
+		transport = &http2.Transport{
+			AllowHTTP: true, // To enable using "http" as protocol
+			DialTLS: func(network, addr string, _ *tls.Config) (net.Conn, error) {
+				// Return the TCP socket without TLS
+				return net.Dial(network, addr)
+			},
+			// TODO: This may not be exactly the same as "MaxResponseHeaderBytes" so check before enabling this
```

Comment:
Does this include restrictions for both response and request? https://datatracker.ietf.org/doc/html/rfc9113#section-10.5.1

Reply:
I am actually not sure at all